workflows/methylation/methylation-cohort.wdl

```diff
@@ -0,0 +1,250 @@
+version 1.1
+
+workflow methylation_cohort {
+    meta {
+        description: "Process methylation data for a cohort of samples"
+        outputs: {
+            combined_beta: "Matrix (in CSV format) containing beta values for every (common) probe on the array as rows and all of the input samples as columns.",
+            filtered_beta: "Matrix (in CSV format) containing only beta values for the retained probes (top N highest standard deviation) for all provided samples.",
+            filtered_probeset: "List of probe names that were retained after filtering to the top N highest standard deviation",
+            umap_embedding: "UMAP embedding for all samples",
+            umap_plot: "UMAP plot for all samples",
+        }
+    }
+
+    parameter_meta {
+        unfiltered_normalized_beta: "Array of unfiltered normalized beta values for each sample"
+    }
+
+    input {
+        Array[File] unfiltered_normalized_beta
+    }
+
+    Int max_length = 500
+    Int beta_length = length(unfiltered_normalized_beta)
+
+    if (beta_length > max_length){
+        scatter (merge_num in range((beta_length / max_length) + 1)){
+            # Get the sublist of beta files
+            scatter (beta_num in range(max_length)){
+                Int num = (
+                    if merge_num > 0
+                    then beta_num + (merge_num * max_length)
+                    else beta_num
+                )
+                if (num < beta_length){
+                    File bam_list = unfiltered_normalized_beta[num]
+                }
+            }
+        }
+        scatter (iter_index in range(length(bam_list))){
+            call combine_data as inner_merge { input:
+                unfiltered_normalized_beta = select_all(bam_list[iter_index]),
+                combined_file_name = iter_index + ".combined.csv",
+                modify_memory_gb = 25,
+            }
+        }
+
+        call combine_data as final_merge { input:
+            unfiltered_normalized_beta = inner_merge.combined_beta,
+        }
+    }
+
+    if (beta_length <= max_length){
+        call combine_data as simple_merge { input:
+            unfiltered_normalized_beta,
+        }
+    }
```

Comment:
What's the deal with the different memory requirements used for these three calls to `combine_data`? Would it be better if `combine_data` dynamically determined memory by the number of inputs or something?

Reply:
To be honest, I don't understand it myself. I suppose I haven't tried the non-iterative merge. But for the iterative merge, the intermediate merge consistently takes more memory. I would like to have additional array cohorts to do some more testing, but these assignments work for now.

Reply:
Ok. Can you leave a comment (or even open an issue) saying as much? "We should do more testing on the memory requirements of `combine_data`"